modules/cats-effect/src/main/scala/pureconfig/module/catseffect/package.scala

```diff
@@ -164,18 +164,28 @@ package object catseffect {
     outputPath: Path,
     blocker: Blocker,
     overrideOutputPath: Boolean = false,
-    options: ConfigRenderOptions = ConfigRenderOptions.defaults())(implicit F: Sync[F], csf: ContextShift[F], writer: Derivation[ConfigWriter[A]]): F[Unit] = {
-    if (!overrideOutputPath && Files.isRegularFile(outputPath))
-      F.raiseError(
-        new IllegalArgumentException(s"Cannot save configuration in file '$outputPath' because it already exists"))
-    else if (Files.isDirectory(outputPath))
-      F.raiseError(
-        new IllegalArgumentException(s"Cannot save configuration in file '$outputPath' because it is a directory"))
-    else
-      Resource.fromAutoCloseable(F.delay(Files.newOutputStream(outputPath))).use { outputStream =>
-        blockingSaveConfigToStreamF(conf, outputStream, blocker, options)
-      }
-  }
+    options: ConfigRenderOptions = ConfigRenderOptions.defaults())(implicit F: Sync[F], csf: ContextShift[F], writer: Derivation[ConfigWriter[A]]): F[Unit] =
+    F.pure(overrideOutputPath).flatMap {
```

Comment:
The old implementation was much easier to understand - we can just wrap it in `F.suspend { ... }` to capture any side effects, and be done with it.

Reply:
I agree it was easier to understand, however, if you look carefully the problem is that some operations are blocking and others not. So after playing with it, I thought this was the best way.

Reply:
Ah, sure, I forgot about blocking. Still, you can see that both `isRegularFile` and `isDirectory` are blocking, so we can use `blocker.delay[F] { ... }.flatten` in a similar fashion.

Reply:
@nigredo-tori That won't work, because that would execute the blocking operations in the compute pool when called `flatten` and would use the blocking context just to construct the inner `F` which is not a blocking operation. I am thinking on alternatives, I will let you know.

Reply:
@nigredo-tori Let me know what do you think of the new implementation.

Reply:
> would execute the blocking operations in the compute pool when called flatten

I'm not quite sure what you meant here, but since we've moved on from that implementation, that doesn't really matter.

> would use the blocking context just to construct the inner F which is not a blocking operation.

I think that's splitting hairs. :smile: But I'm happy with the current implementation (except for the `newOutputStream` thing). I think it does good job with separating path validation from writing the file.